pkg/skaffold/test/custom/custom_test.go

```diff
@@ -0,0 +1,232 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package custom
+
+import (
+	"context"
+	"io/ioutil"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/runcontext"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/GoogleContainerTools/skaffold/testutil"
+)
+
+func TestNewCustomTestRunner(t *testing.T) {
+	testutil.Run(t, "Testing new custom test runner", func(t *testutil.T) {
+		tmpDir := t.NewTempDir().Touch("test.yaml")
+
+		custom := latest.CustomTest{
+			Command:        "echo Running Custom Test command.",
+			TimeoutSeconds: 10,
+			Dependencies: &latest.CustomTestDependencies{
+				Command: "echo [\"file1\",\"file2\",\"file3\"]",
+				Paths:   []string{"**"},
+				Ignore:  []string{"b*"},
+			},
+		}
+
+		cfg := &mockConfig{
+			workingDir: tmpDir.Root(),
+			tests: []*latest.TestCase{{
+				ImageName:      "image",
+				StructureTests: []string{"test.yaml"},
+				CustomTests:    []latest.CustomTest{custom},
+			}},
+		}
+
+		testRunner, err := New(cfg, cfg.workingDir, custom)
+		t.CheckNoError(err)
+		err = testRunner.Test(context.Background(), ioutil.Discard, nil)
+
+		t.CheckNoError(err)
+	})
+}
+
+func TestCustomCommandError(t *testing.T) {
+	tests := []struct {
+		description   string
+		custom        latest.CustomTest
+		shouldErr     bool
+		expectedError string
+	}{
+		{
+			description: "Non zero exit",
+			custom: latest.CustomTest{
+				Command: "exit 20",
+			},
+			shouldErr:     true,
+			expectedError: "exit status 20",
+		},
+		{
+			description: "Command timed out",
+			custom: latest.CustomTest{
+				Command:        "sleep 20",
+				TimeoutSeconds: 2,
+			},
+			shouldErr:     true,
+			expectedError: "context deadline exceeded",
+		},
+	}
+	for _, test := range tests {
+		testutil.Run(t, "Testing new custom test runner", func(t *testutil.T) {
+			tmpDir := t.NewTempDir().Touch("test.yaml")
+
+			if runtime.GOOS == Windows {
+				test.expectedError = "exit status"
+			}
```

Comment:
This is strange and i haven't seen this in test. I think you can get rid of this by overriding the function the `util.RunCmd` and return specific error message or nil error.

Reply:
`util.RunCmd` does not check for the platform.

There are many special cases in the tests of skaffold projects for the platform specific errors. Here is an example:
https://github.com/GoogleContainerTools/skaffold/blob/ba2390db18ad15ccf1eefbce39d6dd8f34016914/pkg/skaffold/kubernetes/portforward/kubectl_forwarder_test.go#L148

Reply:
the point is to not run the command but mock it using overrides to make sure
1) the command is called with right params
2) does the right thing when error or no error is returned.

Reply:
Updated tests to use mock by overriding util.RunCmd method.